File: _정답코드/1주차/10_체스판_다시_칠하기/solution.py
def solution(n, m, board):
    min_repaint = 64
    
    for start_row in range(n - 7):
        for start_col in range(m - 7):
            repaint_w = 0  # 'W'로 시작하는 패턴
            repaint_b = 0  # 'B'로 시작하는 패턴
            
            for i in range(8):
                for j in range(8):
                    current = board[start_row + i][start_col + j]
                    
                    # 'W'로 시작하는 체스판 패턴
                    expected_w = 'W' if (i + j) % 2 == 0 else 'B'
                    if current != expected_w:
                        repaint_w += 1
                    
                    # 'B'로 시작하는 체스판 패턴
                    expected_b = 'B' if (i + j) % 2 == 0 else 'W'
                    if current != expected_b:
                        repaint_b += 1
            
            min_repaint = min(min_repaint, repaint_w, repaint_b)
    
    return min_repaint


n, m = map(int, input().split())
board = []
for _ in range(n):
    board.append(input().strip())

print(solution(n, m, board))

File: _정답코드/1주차/10_체스판_다시_칠하기/solution.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

int solution(int n, int m, vector<string> board) {
    int minRepaint = 64;
    
    for (int startRow = 0; startRow <= n - 8; startRow++) {
        for (int startCol = 0; startCol <= m - 8; startCol++) {
            int repaintW = 0; // 'W'로 시작하는 패턴
            int repaintB = 0; // 'B'로 시작하는 패턴
            
            for (int i = 0; i < 8; i++) {
                for (int j = 0; j < 8; j++) {
                    char current = board[startRow + i][startCol + j];
                    
                    // 'W'로 시작하는 체스판 패턴
                    char expectedW = ((i + j) % 2 == 0) ? 'W' : 'B';
                    if (current != expectedW) repaintW++;
                    
                    // 'B'로 시작하는 체스판 패턴
                    char expectedB = ((i + j) % 2 == 0) ? 'B' : 'W';
                    if (current != expectedB) repaintB++;
                }
            }
            
            minRepaint = min(minRepaint, min(repaintW, repaintB));
        }
    }
    
    return minRepaint;
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0); cout.tie(0);

    int n, m;
    cin >> n >> m;
    
    vector<string> board(n);
    for (int i = 0; i < n; i++) {
        cin >> board[i];
    }
    
    cout << solution(n, m, board) << endl;
    
    return 0;
}

